vendor/src/github.com/docker/engine-api/types/container/config.go

```diff
@@ -5,6 +5,20 @@ import (
 	"github.com/docker/go-connections/nat"
 )
 
+// HealthConfig holds configuration settings for the HEALTHCHECK feature.
+// For durations, a value of -1 means to inherit the default value.
+type HealthConfig struct {
+	Test        strslice.StrSlice // Test to perform to check container is healthy.
+				      // An empty slice means to inherit the default.
+				      // Otherwise, Cmd[0] is "NONE" or "CMD".
+	GracePeriod float64           // Time after container start to assume failure means still starting.
+	Interval    float64           // Time to wait between checks.
+	Timeout     float64           // Time to wait before considering the check to have hung.
+	Retries     uint              // Number of consecutive failures needed to consider a container as unhealthy.
+				      // 0 means default.
+	ExitOnUnhealthy *bool	      `json:",omitempty"` // Stop container on entering unhealthy state. nil for default.
```

Comment:
Is there a reason why this is a nilable option?  They way I see it is that its a yes or no entry and it being nil does not mean anything.  I think it would be better if this could not be nil.